File: src/main/java/ms/aurora/core/Repository.scala
package ms.aurora.core

import scala.collection.mutable


/**
 * A class for holding Session instances
 * @author rvbiljouw
 */
object Repository {
  private val instancesPerGroup = new mutable.HashMap[ThreadGroup, Session]
  private val instancesPerHash =  new mutable.HashMap[Int, Session]

  def get(hashCode: Int): Session = instancesPerHash(hashCode)

  def get(threadGroup: ThreadGroup): Session = instancesPerGroup(threadGroup)

  def set(hashCode: Int, instance: Session) { instancesPerHash += (hashCode -> instance) }

  def set(group: ThreadGroup, instance: Session) { instancesPerGroup += (group -> instance) }

}

File: src/main/java/ms/aurora/core/Repository.java
package ms.aurora.core;

import org.apache.log4j.Logger;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

import static java.lang.Thread.currentThread;

/**
 * @author Rick
 */
public final class Repository {
    private static final Logger logger = Logger.getLogger(Repository.class);
    private static final Map<Integer, Session> sessionMap = new HashMap<Integer, Session>();
    private static final Map<ThreadGroup, Session> groupMap = new HashMap<ThreadGroup, Session>();

    private Repository() {
    }

    public static Session get(Integer appletHash) {
        return sessionMap.get(appletHash);
    }

    public static Session get(ThreadGroup threadGroup) {
        return groupMap.get(threadGroup);
    }

    public static Session set(Integer appletHash, Session session) {
        groupMap.put(currentThread().getThreadGroup(), session);
        return sessionMap.put(appletHash, session);
    }

    public static Collection<Session> getAll() {
        return sessionMap.values();
    }

    public static void foreach(SessionVisitor visitor) {
        for(Session session : getAll()) {
            visitor.visit(session);
        }
    }
}
